### ISSUE TITLE
Cannot Uninstall Openshell Cleanly

### ISSUE THREAD

**Issue by tommykmusic** [Open-Shell/Open-Shell-Menu#1264]
I'm trying to uninstall openshell on windows 11. Tried to uninstall from app and features and nothing worked. Only broke my menu bar when I uninstalled it. Menu Bar didn't change back and the windows Icon on the left broken.

**Comment by ge0rdi** [comment]
> I'm trying to uninstall openshell on windows 11.

What version of `Open-Shell` do you have? What it your Windows 1 version?

> Tried to uninstall from app and features and nothing worked.

Did it show some error message?
Is `Open-Shell` still listed in installed applications?

Did you reboot your machine after uninstall?

> Only broke my menu bar when I uninstalled it.

What exactly do you mean by "menu bar"?
Could you provide some screenshots for clarity?


You can try to install latest version eventually.
And then you may try to uninstall it. Maybe that will help?

**Comment by MichaelJTokarev** [comment]
As with Classic Shell, ­— that too had issues with uninstalling, so a separate utility has been written.

In my case, I installed it on a single machine in a domain (MS AD). Had really difficult time to disable it for all users who can login there. It's really bad when the thing is enabled everywhere *before* there's a chance to disable it.

Now, the admin user which were used to install the thing, has these settings and additional menu in Windows Explorer. On *all* machines in the domain!

So now the question is how to uninstall it from all other machines where it has never been installed but was forced on me? :)

**Comment by ge0rdi** [comment]
> It's really bad when the thing is enabled everywhere before there's a chance to disable it.

You can disable _autostart_ for all users by creating global settings in HKLM (or using group policies).
Please, refer to `Administrative Settings` section in help for details.

> how to uninstall it from all other machines where it has never been installed but was forced on me?

I'm curious how did you achieve to put `Open-Shell` on all machines without installing it there ...

> What's the complete list of places (registry keys and file directories) - system-wide and user-wide - where the Openshell writes itself to?

`Open-Shell` uses following locations:

- `HKEY_LOCAL_MACHINE\SOFTWARE\OpenShell`
- `HKEY_CURRENT_USER\SOFTWARE\OpenShell`
- `%ProgramFiles%\Open-Shell`
- `%APPDATA%\OpenShell`
- `%LOCALAPPDATA%\OpenShell`

**Comment by MichaelJTokarev** [comment]
> You can disable _autostart_ for all users by creating global settings in HKLM (or using group policies). Please, refer to `Administrative Settings` section in help for details.

Yeah I've seen that and added the settings.  I'm not certain here but it seems it works. I still got an explorer toolbar activated - maybe this was from the first attempt (after which I had to uninstall OpenShell, since I found no way to deactivate it for my account once it has been run at least one time).

> > how to uninstall it from all other machines where it has never been installed but was forced on me?
> 
> I'm curious how did you achieve to put `Open-Shell` on all machines without installing it there ...

Once again, this is a domain account which I use to manage multiple computers in a AD domain.  My profile (as is everyone else's) is roaming and being replicated to all other machines. When I log in to any other machine where OpenShell is not installed, I see an active explorer toolbar. I've no idea how it copied itself to other machines - I do see HKU\Software\OpenShell entry, but not how it is enabled in the windows explorer, and where it is being *stored* to begin with.

> > What's the complete list of places (registry keys and file directories) - system-wide and user-wide - where the Openshell writes itself to?
> 
> `Open-Shell` uses following locations:
>     * `HKEY_LOCAL_MACHINE\SOFTWARE\OpenShell`
>     * `HKEY_CURRENT_USER\SOFTWARE\OpenShell`
>     * `%ProgramFiles%\Open-Shell`
>     * `%APPDATA%\OpenShell`
>     * `%LOCALAPPDATA%\OpenShell`

These are the settings for OpenShell itself.  What's missing is the most important part: windows explorer integration, ie, how windows explorer starts OpenShell at the login time (or computer startup time).  This is something I weren't able to find in my case too, for this very toolbar, - I don't see where/how it is started.

BTW, Can't OpenShell be made to install at an account level, not at the system level (as an install choice)? Many programs these days can install either to C:\ProgramFiles or to C:\Users\$USER\AppData\ (or somesuch). When the installer is run with administrative privileges, it offers a choice - to install the software into system or into user; but when it is run without admin privileges, it installs to the user.

Thanks!

**Comment by ge0rdi** [comment]
>  I've no idea how it copied itself to other machines

What folder did you install `Open-Shell` to? Was it standard _program files_ folder? Or your profile one?

> This is something I weren't able to find in my case too, for this very toolbar, - I don't see where/how it is started.

It is registered as shell object. So that _shell_ (explorer) will just load it each time it opens window with some folder.
To unregister it you can use:
`regsvr32 /u <install-dir>\ClassicExplorer64.dll`

Though, I'd suggest to just uninstall `Open-Shell` from your account completely.
That should probably help with the other machines (since installation affected them somehow).

> Can't OpenShell be made to install at an account level, not at the system level (as an install choice)?

This is already tracked in #138.

**Comment by madesitx** [comment]
Hello ! I'm having the exact same issue here. I installed Open Shell, and when I removed it ALL of my domains accounts or new created accounts can't use the start button, search, or even open settings. I have the local admin account that is working, but that's the only one. All the new accounts that will be created have this issue, where I can't open the start button, settings. etc.

Was this ever figured out ?

**Comment by ge0rdi** [comment]
You can try manual uninstall by running our [Utility](https://github.com/Open-Shell/Open-Shell-Menu/releases/download/v4.4.191/Utility.exe) and selecting `Remove Open-Shell`.

Maybe that will help.

**Comment by madesitx** [comment]
That sadly didn't do the trick. I believe Open Shell messed up with the default profile, and now every new profile that gets created will be broken.

I'll post the solution here if I figure it out. Thanks !

**Comment by aryansharmastudies** [comment]
faced a similar sort of issue. I thought the uninstall would be a bod-standard settings/apps/installed apps and uninstall. But upon doing that, my whole wallpaper changed to blue and nothing worked except my cursor. At the time, i also had retrobar installed(perhaps that played into this idk). But, then using ctrl-alt-del and launching task manager i found "windows antivirus" tool to be running(again im not sure if its due to this). To trouble shoot i tried removing it from the startup applications and reboot. However, no success. Then from task manager i opened up cmd prompt trying to create a new user(just incase my current profile becomes forever inaccesable). I found a way to launch firefox from taskmanager(i believe by going into startup apps then find file location and searched up my issue bringing me to this page. I quite luckily installed the Utility program, mentioned 2 ccomments above, and ran it from my file explorer in default settings(ticking none of the 3 boxes).... I suggest uninstalling it using control panel. But otherwise finding a way to download and run the Utility uninstalling service if the situation gets worst. Overall pretty enlightening experience and would totally recommend 10/10. 🥲(jk it was pretty traumatic considering its 12 at night and youre in highschool with all youre desktop being inaccesible)
